File: LuaDemo/Assets/LuaScript/game/service/LoginService.lua
LoginService = BaseService:new();
function LoginService:new()
    local o = BaseService:new();
    setmetatable(o, self);
    self.__index = self;
    return o;
end

local function onGetUser_S(ser, msg)
    Debug.log(msg);
end

function LoginService:onRegist()
    self:loadPb("Login");
    self:registHandler(Cmd.GetUser_S, onGetUser_S, "GetUser_S");
end

function LoginService:getUserInfo(info)
    self:sendData(Cmd.GetUser_C, info, "GetUser_C");
end
return LoginService;

File: ILRuntimeDemo/DemoILRuntime/src/Service/LoginService.cs
using ET;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DemoILRuntime
{
    public class LoginService : BaseService
    {
        private static LoginService _ins;
        public static LoginService Instance
        {
            get
            {
                if (_ins == null) { _ins = new LoginService(); }
                return _ins;
            }
        }

        public override void OnRegist()
        {

            this.RegistCmdHandler(CMD.GetUser_S, OnGetUser_S);
        }

        private void OnGetUser_S(object obj)
        {
            
        }

        public override void OnConnect()
        {
            var msg = new GetUser_C();
            msg.puid = "aabc12";
            this.SendMessage(msg);
        }
        
    }
}
